src/leap/mx/mail_receiver.py

```diff
@@ -188,19 +184,55 @@ def startService(self):
         self.wm = inotify.INotify()
         self.wm.startReading()
 
-        # watch the mail directory for new files and process incoming mail
-        mask = inotify.IN_CREATE
+        # watch mail directories for new files to trigger processing of
+        # incoming mail
         for directory, recursive in self._directories:
-            log.msg("Watching %r --- Recursive: %r" % (directory, recursive))
-            self.wm.watch(filepath.FilePath(directory), mask,
-                          callbacks=[self._process_incoming_email],
-                          recursive=recursive)
+            self._watch_dir(directory, recursive)
 
-        # schedule a periodic task to process skipped mail, but also run it
+        # schedule a periodic task to process skipped mail, and also run it
         # immediatelly
         self._lcall = task.LoopingCall(self._process_skipped)
         self._lcall.start(interval=self.PROCESS_SKIPPED_INTERVAL, now=True)
 
+        # catch SIGUSR1 to trigger processing of skipped mail
+        signal.signal(
+            signal.SIGUSR1,
+            lambda *_: self._process_skipped())
+
+    def stopService(self):
+        """
+        Stop the MailReceiver service
+        """
+        self.wm.stopReading()
+        self._lcall.stop()
+
+    def _watch_dir(self, dirname, recursive):
+        """
+        Start watching a directory to trigger processing of newly created
+        files.
+
+        Will also add a delayed call to retry when failed for some reason.
+        """
+        directory = filepath.FilePath(dirname)
+        try:
+            if not directory.isdir():
```

Comment:
When this conditional is `True` it will raise an `OSError` which will be catched right below where it will be logged.